File: app/src/main/java/com/bopr/android/smailer/util/validator/EmailTextValidator.java
package com.bopr.android.smailer.util.validator;

import android.text.Editable;
import android.widget.EditText;
import android.widget.TextView;

import com.bopr.android.smailer.util.AddressUtil;

/**
 * Checks that {@link EditText}'s input matches email address format.
 *
 * @author Boris Pronin (<a href="mailto:boprsoft.dev@gmail.com">boprsoft.dev@gmail.com</a>)
 */
public class EmailTextValidator extends TextValidator {

    public EmailTextValidator(TextView view) {
        super(view);
    }

    @Override
    public boolean isValidInput(TextView textView, Editable editable, String text) {
        return AddressUtil.isValidEmailAddress(text);
    }

}

File: app/src/main/kotlin/com/bopr/android/smailer/ui/EmailTextValidator.kt
package com.bopr.android.smailer.ui

import android.widget.TextView
import com.bopr.android.smailer.R
import com.bopr.android.smailer.util.isValidEmailAddress

/**
 * Checks that [TextView]'s input matches email address format.
 *
 * @author Boris Pronin ([boprsoft.dev@gmail.com](mailto:boprsoft.dev@gmail.com))
 */
class EmailTextValidator(view: TextView) : TextValidator(view) {

    override fun getErrorMessage(text: String?): String? {
        return when {
            text.isNullOrEmpty() ->
                null
            !isValidEmailAddress(text) ->
                view.context.getString(R.string.invalid_email)
            else ->
                null
        }
    }
}